codemp/game/g_cmds.cpp

```diff
@@ -2987,6 +2993,11 @@ void SanitizeString2( char *in, char *out )
 				i += 2;
 				continue;
 			}
+			else if (in[i + 1] == 'x' || in[i + 1] == 'X')		//if an extended RGB color code
+			{
+				i += 5;
```

Comment:
You're at risk of reading off the end of the string if you do this. Imagine if a string ends with "^x", and then you increment `i` by 5, The `while` condition will try to read off the end off the string and can cause all kinds of badness.

You also can't assume that the three characters following the ^x will be valid characters to form an RGB color.